src/main/java/sirius/web/data/XLSProcessor.java

```diff
@@ -58,6 +66,42 @@ public void run(RowProcessor rowProcessor, Predicate<Exception> errorHandler) th
         }
     }
 
+    /**
+     * Starts processing the given sheet and sends each line to the given sheetProcessor.
+     *
+     * @param sheetProcessor the processor which handles each row of a sheet
+     * @param errorHandler   errorHandler which gets called on exceptions. returns <tt>true</tt> if the exception was
+     *                       handled and <tt>false</tt> if the exception should be rethrown.
+     * @param sheetNames     the sheet names to process
+     * @throws Exception                             in case an error occurred while processing.
+     * @throws IOException                           if the stream providing the given workbook cannot be read.
+     * @throws sirius.kernel.health.HandledException if workbook does not contain a sheet with the given name.
+     */
+    public void runForSheets(SheetBasedRowProcessor sheetProcessor,
+                             Predicate<Exception> errorHandler,
+                             String... sheetNames) throws Exception {
```

Comment:
I'd change this to runForSheets(SheetBasedRowProcessor... processors)

And make SheetProcessor a simple class which contains a sheetName, a RowProcessor and an error Handler Predicate<Exception>